lib/puppet/application/lookup.rb

```diff
@@ -339,6 +339,7 @@ def generate_scope
 
     Puppet[:code] = 'undef' unless options[:compile]
     compiler = Puppet::Parser::Compiler.new(node)
+    compiler.loaders # ensures that Pcore et.al. is initialized
```

Comment:
Is there a way to ensure pp code has been loaded when the compiler is created? It just seems strange that we create a compiler object, but it's not fully functional.